runtime/bundles/org.eclipse.core.jobs/src/org/eclipse/core/runtime/jobs/IJobRunnable.java

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Christoph Läubrich and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Christoph Läubrich - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.core.runtime.jobs;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+
+/**
+ * A pice of coude that runs inside a job and has access to a progressmonitor
+ *
+ * @since 3.14
+ *
+ */
+public interface IJobRunnable {
```

Comment:
Can't we reuse the existing `ICoreRunnable` instead of this new interface with very similar signature and only different exception that may be thrown? In general it's bit unfortunate that Eclipse has invented it's own 'root' exception, this make integration with standard Java methods often cumbersome.

We could then just add only a new `ICoreCallable`? There we could also throw just `Exception`.
Then there would certainly be a asymmetry between `ICoreRunnable` and `ICoreCallable`, but in my opinion the big value of this change is the new `Job.compute()` method, so I assume it's used much more often.